src/main/java/com/nha/abdm/wrapper/hiu/hrp/consent/ConsentGatewayCallbackService.java

```diff
@@ -56,18 +56,18 @@ public HttpStatus onInitConsent(OnInitRequest onInitRequest) throws IllegalDataS
     if (Objects.nonNull(onInitRequest)
         && Objects.nonNull(onInitRequest.getResp())
         && Objects.nonNull(onInitRequest.getConsentRequest())) {
-      requestLogService.updateConsentResponse(
-          onInitRequest.getResp().getRequestId(),
-          FieldIdentifiers.CONSENT_ON_INIT_RESPONSE,
-          RequestStatus.CONSENT_ON_INIT_RESPONSE_RECEIVED,
-          onInitRequest.getConsentRequest().getId());
       // This mapping needs to be persisted in database because when gateway issues hiu notify call,
       // it passes
       // consent request id and then there is no way to track original request other that looping
       // through all the requests
       // and checking their responses for consentRequestId.
       consentRequestService.saveConsentRequest(
-          onInitRequest.getConsentRequest().getId(), onInitRequest.getResp().getRequestId());
+              onInitRequest.getConsentRequest().getId(), onInitRequest.getResp().getRequestId());
+      requestLogService.updateConsentResponse(
```

Comment:
what was the motivation behind moving this line of code?

Reply:
There is huge change in abdm internal workings, all the API goes through the transformer and it goes to respective v2 or v3 systems. because of that if there is any error example:/consents/init ->carecontexts are not present the abdm sends /on-init and also /hiu/notify at the same time, in hiu notify it throws an error, since both are coming at the same time, hiu/notify is searching for consentRequestId and the consentRequestId storing is in the process ,because of the updateConsentResponse(), to avoid that I have moved that function above.